src/renderer/drawers/Provider.js

```diff
@@ -0,0 +1,51 @@
+/* @flow */
+import React, { useReducer } from "react";
+import { SideDrawer } from "~/renderer/components/SideDrawer";
+
+type State = {
+  Component: *,
+  props: *,
+  open: boolean,
+};
+
+// actions
+// it makes them available and current from connector events handlers
+export let setDrawer: (Component?: *, props?: *) => void = () => {};
+
+// reducer
+const reducer = (state: State, update) => {
+  return {
+    ...state,
+    ...update,
+  };
+};
+const initialState = {
+  Component: null,
+  props: null,
+  open: false,
+};
+
+export const context = React.createContext<State>(initialState);
```

Comment:
You export the context here but dont seem to use it anywhere :shrug:

Reply:
its used in the root of the app, in Default

Reply:
updated i ve extracted the drawer component